src/dbnode/persist/fs/seek.go

```diff
@@ -380,16 +374,9 @@ func (s *seeker) SeekIndexEntry(
 		return IndexEntry{}, err
 	}
 
-	seekedOffset, err := s.indexFd.Seek(offset, os.SEEK_SET)
-	if err != nil {
-		return IndexEntry{}, err
-	}
-	if seekedOffset != offset {
-		return IndexEntry{}, instrument.InvariantErrorf(
-			"tried to seek to offset: %d, but seeked to: %d", seekedOffset, offset)
-	}
-
-	resources.fileDecoderStream.Reset(s.indexFd)
+	resources.offsetFileReader.reset(s.indexFd)
+	resources.offsetFileReader.setOffset(offset)
+	resources.fileDecoderStream.Reset(resources.offsetFileReader)
```

Comment:
tbh here's where I mean it'd be easier to avoid reusing the offsetFileReader and just alloc a new one. don't think it's going to cause any performance issues either way.

Reply:
There are users I see doing tens of thousands of reads per second, might be worth continuing to keep per operation allocation low/zero where possible.